interfaces/cython/cantera/examples/onedim/diffusion_flame_batch.py

```diff
@@ -27,10 +28,17 @@ class FlameExtinguished(Exception):
     pass
 
 
-# Create directory for output data files
-data_directory = 'diffusion_flame_batch_data/'
-if not os.path.exists(data_directory):
-    os.makedirs(data_directory)
+hdf_output = find_spec('h5py') is not None
+
+if not hdf_output:
+    # Create directory for output data files
+    data_directory = 'diffusion_flame_batch_data/'
+    if not os.path.exists(data_directory):
+        os.makedirs(data_directory)
+    fig_name = '{}figure_{{0}}.png'.format(data_directory)
```

Comment:
In this case, I find this simpler:
```suggestion
    fig_name = data_directory + 'figure_{0}.png'
```

Reply:
This may also be a good place to use `pathlib`, there might be a few places you have to cast the `Path` instance to a `str()` though, in Cantera's methods. I haven't tried it though, depending on how things are done internally, it may "just work", but then again, maybe only on Python 3.6+ which is I believe when they switched over to the built-in `open()` allowing `pathlib`.

Reply:
I opted to replace the path concatenations by `os.path.join`, i.e. keeping strings albeitin a 'cleaner' approach. Refactoring how paths are handled isn't the main priority here, especially as there's the possibility of dropping XML from Python altogether.